references/recaptcha-enterprise/pipeline.sh

```diff
@@ -0,0 +1,324 @@
+                        {
+                            "name": "GCP_PROJECT_ID",
+                            "value": "$GCP_PROJECT_ID"
+                        },
+                        {
+                            "name": "SITE_KEY",
+                            "value": "$SITEKEY_ALWAYS_0"
+                        }
+                    ],
+                    "apiProducts": [
+                        "RecaptchaEnterprise"
+                    ],
+                    "callbackUrl": "",
+                    "scopes": []
+                }
+            ]
+        }
+    }
+}
+EOF
+}
+
+#################################
+### function: set_idp_env_var ###
+#################################
+set_functional_test_env_var() {
+
+    # use timestamp (parameter) to create a unique value
+    TEST_APP_CONSUMER_KEY="xkey-$1"
+    export TEST_APP_CONSUMER_KEY
+    TEST_APP_CONSUMER_SECRET='xsecret'
+    export TEST_APP_CONSUMER_SECRET
+}
+
+######################################
+### function: set_recaptcha_config ###
+######################################
+set_recaptcha_config() {
+
+    # Copy AM policy template
+    cp "$SCRIPTPATH"/templates/AM-SetReCaptchaMock.template.xml "$SCRIPTPATH"/sf-recaptcha-enterprise-v1/sharedflowbundle/policies/AM-SetReCaptchaMock.xml
+
+    # replace reCAPTCHA tag in AM policy
+    sed -i.bak "s|@IsReCaptchaMockEnabled@|$1|" "$SCRIPTPATH"/sf-recaptcha-enterprise-v1/sharedflowbundle/policies/AM-SetReCaptchaMock.xml
+
+    # remove .bak files
+    rm "$SCRIPTPATH"/sf-recaptcha-enterprise-v1/sharedflowbundle/policies/AM-SetReCaptchaMock.xml.bak
+}
+
+### script execution starts here...
+
+# Check for required variables
+if [ -z "$GCP_PROJECT" ]; then
+  echo "The required env variable GCP_PROJECT is missing";
+  exit 1
+fi
+
+# Manage optional variables
+GCP_REGION=${GCP_REGION:-europe-west1}
+timestamp=$(date '+%s')
+export PATH="$PATH:$SCRIPTPATH/../../tools/apigee-sackmesser/bin"
+
+# default value for reCAPTCHA mock mode (default is 'true')
+DEFAULT_IS_RECAPTCHA_MOCK_ENABLED=true
+
+# is reCAPTCHA mock enabled (=true) or not (=false)
+IS_RECAPTCHA_MOCK_ENABLED="${IS_RECAPTCHA_MOCK_ENABLED:-"$DEFAULT_IS_RECAPTCHA_MOCK_ENABLED"}"
+
+# set reCAPTCHA config
+set_recaptcha_config "$IS_RECAPTCHA_MOCK_ENABLED"
+
+echo "[INFO] Deploying reCAPTCHA enterprise reference to Google API (For X/hybrid)"
+APIGEE_TOKEN=$(gcloud auth print-access-token);
+
+## Generate Service Account for Apigee shared flow
+gcloud iam service-accounts create apigee-recaptcha-enterprise-sa \
+--project "$GCP_PROJECT" || true
+
+# Create the service account that is used to invoke the Google reCAPTCHA enterprise endpoint
+gcloud projects add-iam-policy-binding "$GCP_PROJECT" \
+    --member="serviceAccount:apigee-recaptcha-enterprise-sa@$GCP_PROJECT.iam.gserviceaccount.com" \
+    --role="roles/recaptchaenterprise.agent"
+
+# If reCAPTCHA mock is enabled
+if [ -z "$IS_RECAPTCHA_MOCK_ENABLED" ] || [ "$IS_RECAPTCHA_MOCK_ENABLED" = "true" ];then
```

Comment:
`IS_RECAPTCHA_MOCK_ENABLED` should never be empty if you set it to true if missing

Reply:
Yes, right